tally_ho/apps/tally/models/audit.py

```diff
@@ -74,6 +74,90 @@ def action_prior_name(self):
     def resolution_recommendation_name(self):
         return _(self.resolution_recommendation.label)
 
+    def get_quarantine_check_details(self, check):
+        """Get the details of a failed quarantine check
+        including actual values.
+
+        :param check: The QuarantineCheck instance
+        :returns: A dictionary with check details and actual values
+        """
+        result_form = self.result_form
+        recon_form = getattr(result_form, 'reconciliationform', None)
+
+        if not recon_form:
+            return None
+
+        details = {
+            'name': check.local_name(),
+            'tolerance_value': check.value,
+            'tolerance_percentage': check.percentage,
+        }
+
+        if check.method == 'pass_reconciliation_check':
+            expected_total = (
+                result_form.num_votes +
+                recon_form.number_invalid_votes
+            )
+            actual_total = recon_form.number_sorted_and_counted
+            allowed_tolerance = (
+                check.value if check.value != 0
+                else ((check.percentage / 100) * expected_total)
+            )
```

Comment:
This may be a duplicate calculation, which would mean a change in one place needs to be applied to the other. Quarantine checks/calculations have changed between the seasons of elections. I would expect them to change. Easy for someone to forget to update all places.